apps/api/src/controllers/collaboratorsController.ts

```diff
@@ -0,0 +1,100 @@
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { getDbInstance } from '@/db/index.js';
+import { applicationSvc } from '@/service/applicationService.js';
+import { collaboratorsSvc } from '@/service/collaboratorsService.js';
+import { type ApplicationService, type CollaboratorModel, type CollaboratorsService } from '@/service/types.js';
+import { failure } from '@/utils/results.js';
+import { CollaboratorDTO } from '@pcgl-daco/data-model';
+
+/* Type to assert Collaborator record has valid required fields */
+interface ValidCollaboratorDTO extends CollaboratorDTO {
+	collaboratorFirstName: string;
+	collaboratorLastName: string;
+	collaboratorPositionTitle: string;
+	collaboratorInstitutionalEmail: string;
+}
+
+/**
+ * Creates a new collaborator and returns the created data.
+ * @param application_id - ID of related application record to associate with Collaborators
+ * @param user_id - ID of Applicant updating the application
+ * @param collaborators - Array of new Collaborators to create
+ * @returns Success with Collaborator data array / Failure with Error.
+ */
+export const createCollaborators = async ({
+	application_id,
+	user_id,
+	collaborators,
+}: {
+	application_id: number;
+	user_id: string;
+	collaborators: CollaboratorDTO[];
+}) => {
+	const database = getDbInstance();
+	const collaboratorsRepo: CollaboratorsService = collaboratorsSvc(database);
+	const applicationRepo: ApplicationService = applicationSvc(database);
+
+	const applicationResult = await applicationRepo.getApplicationById({ id: application_id });
+
+	if (!applicationResult.success) {
+		return applicationResult;
+	}
+
+	const application = applicationResult.data;
+
+	// TODO: Add Real Auth
+	// Validate User is Applicant
+	if (!(user_id === application.user_id)) {
+		return failure('Unauthorized, cannot create Collaborators');
+	}
+
+	if (!(application.state === 'DRAFT')) {
+		return failure(`Can only add Collaborators when Application is in state DRAFT`);
+	}
+
+	const validCollaborators: ValidCollaboratorDTO[] = collaborators.filter(
+		(data: CollaboratorDTO): data is ValidCollaboratorDTO =>
+			!!data.collaboratorFirstName &&
+			!!data.collaboratorLastName &&
+			!!data.collaboratorPositionTitle &&
+			!!data.collaboratorInstitutionalEmail,
+	);
+
+	if (!(validCollaborators.length === collaborators.length)) {
+		return failure('Required Collaborator details are missing.');
+	}
+
+	const newCollaborators: CollaboratorModel[] = validCollaborators.map((data) => ({
+		first_name: data.collaboratorFirstName,
+		middle_name: data.collaboratorMiddleName,
+		last_name: data.collaboratorLastName,
+		suffix: data.collaboratorSuffix,
+		position_title: data.collaboratorPositionTitle,
+		institutional_email: data.collaboratorInstitutionalEmail,
+		application_id,
+	}));
+
+	const result = await collaboratorsRepo.createCollaborators({
+		newCollaborators,
+	});
```

Comment:
Do we need to do any checks for duplicate collaborators?

Reply:
Wondered the same thing

Reply:
Added handling for preventing duplicates in req.body, now adding db layer

Reply:
OK I've added logic to prevent creating duplicates in the database
Required quite a few changes which is why I didn't finish last night:
[Prevenet Duplicate Collaborator Records](https://github.com/Pan-Canadian-Genome-Library/daco/pull/175/commits/0b58ea4a76efa22ccdc441d2b8ea6ef1d68520c3)